### ISSUE TITLE
Cannot cancel compiling script directory

### ISSUE THREAD

**Issue by astrelsky** [NationalSecurityAgency/ghidra#3055]
**Describe the bug**
Attempting to cancel script directory compilation when running a script for the first time doesn't work.

**Environment (please complete the following information):**
 - Java Version: 15.0.2
 - Ghidra Version: 10.0_DEV 70675fce99a4c6e6e650729e5dda6ccbbbbbd40d

**Comment by jpleasu** [comment]
There are two obstacles, maybe three, to respecting that cancel button.
1) the age-old API `GhidraScriptProvider` never gets a monitor.  Namely, `getScriptInstance` which is the typical path to (re)compilation.
2) although every call to `GhidraBundle#build` _might_ get a monitor, it doesn't.
3) We gauged the necessity of interruption by the longest typical run, which seemed to be  compiling the scriptdir of  `Features/Base` - and that takes a couple of seconds on an a low/average spec machine.

1&2 are are "just code" but 3..  it's either wrong or your use is unsupported.

If we got the timing wrong for normal usage, we can fix it.

A couple of things that should help:
1) don't leave broken scripts in an active script directory.  The back-off strategy will try to ignore scripts that have failed, but they're always tried at least once per Ghidra run.
2) split up your code and use `@importpackage`.
